### DISCUSSION TITLE
Please add touch select support to 3D panel

### DISCUSSION THREAD

**Discussion by kqhasaki** [foxglove/community#652] [Q&A]
Recently we are working on a project based on foxglove and intending to make a customized HMI tool based on 3D panel. We have noticed that object selection in 3D panel can only be achieved through mouse click while it seems to be some unfinished support for touch event in the code base. Could this support for touch event be finished sooner as a feature for object selection through touch?

```typescript

  private onTouchStart = (event: TouchEvent): void => {
    const touch = event.touches[0];
    if (touch) {
      this.startClientPos = new THREE.Vector2(touch.clientX, touch.clientY);
    }
    event.preventDefault();
  };

  private onTouchEnd = (event: TouchEvent): void => {
    event.preventDefault();
  };

  private onTouchMove = (event: TouchEvent): void => {
    event.preventDefault();
  };

  private onTouchCancel = (event: TouchEvent): void => {
    event.preventDefault();
  };
```

**Comment by jtbandes** [comment]
Hi, would you be interested in contributing a PR to implement this feature?

**Reply by Pana1v10x** [reply]
hi, I'm Interested, Ive already implemented this on lichtblick
